src/typescript/frontend/src/components/OrderBook.tsx

```diff
@@ -0,0 +1,175 @@
+import { Listbox } from "@headlessui/react";
+import { ChevronDownIcon } from "@heroicons/react/20/solid";
+import { useQuery, type UseQueryResult } from "@tanstack/react-query";
+import { useEffect, useRef, useState } from "react";
+
+import { type ApiMarket } from "@/types/api";
+
+type OrderPrice = {
+  price: number;
+  size: number;
+};
+
+type OrderBook = {
+  bids: OrderPrice[];
+  asks: OrderPrice[];
+};
+
+const PrecisionOptions = ["0.01", "0.05", "0.1", "0.5", "1", "2.5", "5", "10"];
+
+export function OrderBook({ marketData }: { marketData: ApiMarket }) {
+  const [asks, setAsks] = useState<OrderPrice[]>([]);
+  const [bids, setBids] = useState<OrderPrice[]>([]);
+  const [spread, setSpread] = useState<OrderPrice>();
+  const [highestSize, setHighestSize] = useState<number>(0);
+  const [precision, setPrecision] = useState<string>(PrecisionOptions[0]);
+  const [initialRender, setInitialRender] = useState<boolean>(false);
+  const data = useOrderBook("", precision);
```

Comment:
Personally I prefer having all the data fetching logic inside of the page and passing the data down to components as props, so you can see every API a page is making in a single glance, but I’ve left that as is for now in the review PR

Reply:
decided to leave it into a useOrderbook hook